src/index.ts

```diff
@@ -41,57 +43,31 @@ export class I18N {
     }
 
     has(keysetName: string, key: string, lang?: string) {
-        const languageData = this.getLanguageData(lang);
+        const languageData = this.getLanguageData(lang || this.lang);
 
         return Boolean(languageData && languageData[keysetName] && languageData[keysetName][key]);
     }
 
     i18n(keysetName: string, key: string, params?: Params): string {
-        const languageData = this.getLanguageData(this.lang);
-
-        if (typeof languageData === 'undefined') {
-            throw new Error(`Language '${this.lang}' is not defined, make sure you call setLang for the same language you called registerKeysets for!`);
-        }
-
-        if (Object.keys(languageData).length === 0) {
-            this.warn('Language data is empty.');
-
-            return key;
-        }
+        this.checkLangDefinitionOrThrow();
 
-        const keyset = languageData[keysetName];
-
-        if (!keyset) {
-            this.warn(
-                'Keyset not found.',
-                keysetName,
-            );
+        let keyValue = this.getValue(keysetName, key, this.lang);
 
-            return key;
-        }
-
-        if (Object.keys(keyset).length === 0) {
-            this.warn(
-                'Keyset is empty.',
-                keysetName,
-            );
+        if (!keyValue) {
+            if (this.lang === this.defaultLang || !this.defaultLang) {
+                return key;
+            }
+            
+            this.warn(`Switch to default language ${this.defaultLang}`);
 
-            return key;
+            keyValue = this.getValue(keysetName, key, this.defaultLang);
```

Comment:
Second call of `this.getValue` will trigger the same warning logs as the first one, this could be annoying. Can we check for default lang inside the `getValue` and not call it two times?
Consider this interface:
```ts
getValue(keysetName: string, key: string, lang?: string, defaultLang?: string);
getKeysetData(keysetName: string, lang?: string, defaultLang?: string);
getLanguageData(lang?: string, defaultLang?: string);
```

Reply:
Second messege inside `this.getValue` will be diplayed after `Switch to default language ...`, message. It seems that this information is enough.